### PR TITLE
Google Maps in app on iOS

### PR DESCRIPTION
**Pull request by markholland**
We will need to use Google Maps in the app if we want to overlay information.

We used CocoaPods to assist with the initial setup and have decided to go with a recommended approach of checking in the Pods folder. This means we won't have to introduce Cocoapods to our dev or CI workflows which can cause headaches around versioning and greatly increase CI build time.

## Pre-merge check-list

* [ ] Link to Trello ticket/GitHub issue (If applicable)
* [ ] Any risky areas identified
* [ ] Test plan provided
* [ ] Tester approved

## Tester check-list

* [ ] Tested on multiple devices / OS versions (Test on the physical device(s) you have access to, otherwise use BrowserStack):

  * [ ] Galaxy S8 (Android 7.0/8.0)
  * [ ] Galaxy S7 (Android 6.0)
  * [ ] iPhone X (iOS 11.x)
  * [ ] iPhone 8 (11.x)
  * [ ] iPhone 7 (iOS 10.x)

  Optional:

  * [ ] Google Pixel 2
  * [ ] Galaxy S8 Plus
  * [ ] Galaxy S7 Edge
  * [ ] Galaxy S6
  * [ ] iPhone 7+
  * [ ] iPhone 6

Accessibility Testing (If applicable)

* [ ] Text-to-Voice (Android: Voice Assistant / iOS: Speak Selection)
* [ ] Large Text (=<200%)
* [ ] Landscape Screen orientation

Weak connection testing (If applicable)

* [ ] Airplane mode
* [ ] 2G/3G Network Settings

### CODE REVIEW

File: ios/Podfile [hunk 1]
```diff
@@ -0,0 +1,6 @@
+platform :ios, '9.0'
```
Comment (markholland) [inline ios/Podfile]: This is our Podfile which we can add to

File: src/screens/EventDetailsScreen/EventMap.js [hunk 2]
```diff
@@ -19,6 +19,7 @@ const EventMap = ({ lat, lon, locationName }: Props) => (
   >
     <Delayed delay={600}>
       <MapView
+        provider={PROVIDER_GOOGLE}
```
Comment (markholland) [inline src/screens/EventDetailsScreen/EventMap.js]: We use Google Maps always

File: ios/PrideLondonApp/AppDelegate.m [hunk 3]
```diff
@@ -28,6 +31,8 @@ - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(
   
   [BugsnagReactNative start];
   
+  [GMSServices provideAPIKey:[ReactNativeConfig envFor:@"GOOGLE_MAPS_API_KEY"]];
```
Comment (markholland) [inline ios/PrideLondonApp/AppDelegate.m]: We set the API key for Google Maps from our .env

**Comment by robbiemccorkell** [comment]
I was never sure what to do with the cocoapods files. Is it normal to just commit them all, or do people usually have an install step in their build pipeline?

**Comment by markholland** [comment]
@robbiemccorkell http://guides.cocoapods.org/using/using-cocoapods.html#benefits-of-checking-in-the-pods-directory

**Comment by declanslevin** [comment]
The maps preview on the Event details page only loads correctly on the first Event opened, thereafter it appears as a pale blue colour:

**Comment by markholland** [comment]
Great https://github.com/react-community/react-native-maps/issues/1452

**Comment by markholland** [comment]
@declanslevin should be fixed in more recent builds

**Comment by markholland** [comment]
I'll want to try and do this without CocoaPods as it has significantly increased build times. But this can be a back up should we need to start developing against Google Maps.

**Comment by robbiemccorkell** [comment]
Out of interest, how do we know that overlays aren't supported in Apple Maps by react native maps? I hadn't seen that limitation.

**Comment by markholland** [comment]
My understanding was that the overlay would be created in a google maps account and then the app would just refer to it.

**Comment by markholland** [comment]
No need to merge this until we need it of course

**Comment by declanslevin** [comment]
Retested on multiple iOS devices, map previews are appearing correctly. Looks good, I won't merge this for now though due to the previous comments.

**Comment by markholland** [comment]
Have spoken with @frigus02 and this hopefully won't be needed. We should definitely try and use just the capabilities of react-native-maps. Also we should look into a feature of google developer console where you don't have to embed the google maps api key in the app. Would mean any key revokes wouldn't require app updates.

Will close this PR but keep the branch

**Pull request closed without merging**